# GA0029

## Title
Invalid mutually exclusive declarations of the interface

## Category
Validity

## Severity
Error

## Details
This error is emitted when an interface is declared as both a profile and a profile group.

## Example
```csharp
// GA0029 will appear here
[TypeConstraintProfile]
[TypeConstraintProfileGroup]
interface IProfileAndGroup { }
```