# GA0027

## Title
Invalid type constraint profile group type

## Category
Validity

## Severity
Error

## Details
This error is emitted when a type that is not a type constraint profile group type is assigned in context where a type constraint profile group type is expected. This applies to assigning type constraint profile interface to profile groups.

Any non-interface type, or interface type that is not assigned as a type constraint profile will yield this error when used in the relevant context.

## Example
```csharp
// GA0027 will appear here
[TypeConstraintProfile(typeof(IGroup), typeof(IUnrelated))]
interface IProfile { }

[TypeConstraintProfileGroup]
interface IGroup { }

interface IUnrelated { }
```
