# GA0023

## Title
Invalid type constraint profile type declaration

## Category
Validity

## Severity
Error

## Details
This error is emitted when a type has been declared as a type constraint profile (group), but is not an interface.

Only interface types are eligible for type constraint profiles and type constraint profile groups.

## Example
```csharp
// GA0023 will appear here
[TypeConstraintProfile]
class A { }

// GA0023 will appear here
[TypeConstraintProfileGroup]
struct B { }

// GA0023 will appear here
[TypeConstraintProfile]
delegate void C();

// GA0023 will appear here
[TypeConstraintProfile]
enum D { }
```
