# GA0026

## Title
Invalid type constraint profile type

## Category
Validity

## Severity
Error

## Details
This warning is emitted when a type that is not a type constraint profile type is assigned in context where a type constraint profile type is expected. This applies to inheriting profile type constraints from a type constraint profile interface.

Any non-interface type, or interface type that is not assigned as a type constraint profile will yield this error when used in the relevant context.

## Example
```csharp
[TypeConstraintProfile]
interface IProfile { }

interface IUnrelated { }

class C
<
    // GA0026 will appear here
    [InheritProfileTypeConstraints(typeof(IUnrelated))]
    T,
    [InheritProfileTypeConstraints(typeof(IProfile))]
    U
> { }
```
